src/core/dev_api/shape_util.hpp

```diff
@@ -26,5 +26,11 @@ OPENVINO_API Shape make_dynamic_shape();
  */
 OPENVINO_DEPRECATED("This function is deprecated and will be removed soon.")
 OPENVINO_API bool is_dynamic_shape(const Shape& s);
+
+OPENVINO_API Shape reduce(const Shape& input, const AxisSet& axes);
+OPENVINO_API Shape reduce(const Shape& input, const AxisSet& axes, const bool keep_dims);
+OPENVINO_API std::vector<size_t> reduce(const std::vector<size_t>& input, const AxisSet& axes);
+
+OPENVINO_API Shape reduce_keep_dims(const Shape& input, const AxisSet& axes);
```

Comment:
I don't see that anybody uses `reduce_keep_dims` method outside the common library, so looks like it can be removed from the developer API

Reply:
The plan was to replace it for example in [mvn refernce](https://github.com/openvinotoolkit/openvino/blob/25b1b4e26c265bd0c11f52c34ba45bd69bb1e78c/src/core/reference/include/openvino/reference/mvn.hpp#L29).

I can hide but where should I put declaration to be available in reference implementaion?

And just maybe as TODO in separate PR. 
Should the `shape_utils.hpp` be moved into sub-folder `openvino/core` in dev_api?

Reply:
Yes it will be better to move to sub folder